src/components/auth/profile/UpdateEmailModal.tsx

```diff
@@ -0,0 +1,146 @@
+import GenericForm from 'components/common/form/GenericForm'
+import SubmitButton from 'components/common/form/SubmitButton'
+import FormTextField from 'components/common/form/FormTextField'
+import { Person, UpdateUserAccount, UpdatePerson } from 'gql/person'
+import { useMutation } from 'react-query'
+import { AxiosError, AxiosResponse } from 'axios'
+import { ApiErrors } from 'service/apiErrors'
+import { updateCurrentPerson } from 'common/util/useCurrentPerson'
+import { AlertStore } from 'stores/AlertStore'
+import { useTranslation } from 'next-i18next'
+import CloseIcon from '@mui/icons-material/Close'
+import * as yup from 'yup'
+import { email, password } from 'common/form/validation'
+import { signIn, signOut } from 'next-auth/react'
+import PasswordField from 'components/common/form/PasswordField'
+import { useState } from 'react'
+import { baseUrl, routes } from 'common/routes'
+
+const PREFIX = 'UpdateNameModal'
+
+const classes = {
+  modal: `${PREFIX}-modal`,
+  close: `${PREFIX}-close`,
+}
+
+const StyledModal = styled(Modal)(({ theme }) => ({
+  [`& .${classes.modal}`]: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: 650,
+    backgroundColor: '#EEEEEE',
+    padding: 20,
+    [theme.breakpoints.down('md')]: {
+      width: '70%',
+    },
+  },
+  [`& .${classes.close}`]: {
+    position: 'absolute',
+    right: '10px',
+  },
+}))
+
+const validationSchema: yup.SchemaOf<Pick<UpdateUserAccount, 'email' | 'password'>> = yup
+  .object()
+  .defined()
+  .shape({
+    email: email.required(),
+    password: password.required(),
+  })
+
+const callbackUrl = `${baseUrl}${routes.login}`
+
+function UpdateEmailModal({
+  isOpen,
+  handleClose,
+  person,
+}: {
+  isOpen: boolean
+  handleClose: (data?: Person) => void
+  person: UpdatePerson
+}) {
+  const { t } = useTranslation()
+  const [loading, setLoading] = useState(false)
+
+  const mutation = useMutation<AxiosResponse<Person>, AxiosError<ApiErrors>, UpdatePerson>({
+    mutationFn: updateCurrentPerson(),
+    onError: () => AlertStore.show(t('common:alerts.error'), 'error'),
+    onSuccess: () => AlertStore.show(t('common:alerts.success'), 'success'),
+  })
+
+  const onSubmit = async (values: Pick<UpdateUserAccount, 'email' | 'password'>) => {
+    try {
+      setLoading(true)
+
+      const confirmPassword = await signIn<'credentials'>('credentials', {
+        email: person.email,
+        password: values.password,
+        redirect: false,
+      })
+      if (confirmPassword?.error) {
+        handleClose()
+        AlertStore.show(t('auth:alerts.invalid-login'), 'error')
+        throw new Error('Invalid login')
+      }
+
+      const updateUser = await mutation.mutateAsync({
+        ...person,
+        email: values.email,
+      })
+
+      const reLogin = await signIn<'credentials'>('credentials', {
+        email: person.email,
+        password: values.password,
+        redirect: false,
```

Comment:
It looks the reLogin here should happen with the new email value, like this: email: values.email